### PR TITLE
Add multiple service accounts for pipeline start

### PR DESCRIPTION
**Pull request by khrm**
Add task specific multiple service accounts for pipeline start.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

File: pkg/cmd/pipeline/start.go [hunk 1]
```diff
@@ -259,3 +294,20 @@ func parseParam(p []string) (map[string]v1alpha1.Param, error) {
...
 	return params, nil
 }
+
+func parseSvc(s []string) (map[string]v1alpha1.PipelineRunSpecServiceAccount, error) {
+	svcs := map[string]v1alpha1.PipelineRunSpecServiceAccount{}
+	for _, v := range s {
+		r := strings.Split(v, "=")
+		if len(r) != 2 || len(r[0]) == 0 {
+			errMsg := invalidSvc + v +
+				"\n Please pass service accounts as -r TaskName=ServiceAccount"
```
Comment (hrishin) [inline pkg/cmd/pipeline/start.go]: nit: remove the space `\n Please` after new line.

How about `Please pass service accounts as -r TaskName=ServiceAccount` -> `Please pass service accounts as TaskName=ServiceAccount`?

File: pkg/cmd/pipeline/start.go [hunk 2]
```diff
@@ -197,6 +207,31 @@ func mergeParam(pr *v1alpha1.PipelineRun, optPar []string) error {
...
+	}
+
+	if len(svcs) == 0 {
+		return nil
+	}
+
+	for i := range pr.Spec.ServiceAccounts {
+		if v, ok := svcs[pr.Spec.ServiceAccounts[i].TaskName]; ok {
+			pr.Spec.ServiceAccounts[i] = v
+			delete(svcs, v.TaskName)
```
Comment (hrishin) [inline pkg/cmd/pipeline/start.go]: wondering do we really need this delete call?

File: pkg/cmd/pipeline/start_test.go [hunk 3]
```diff
@@ -572,6 +582,47 @@ func Test_parseParam(t *testing.T) {
 	}
 }
 
+func Test_parseSvc(t *testing.T) {
```
Comment (hrishin) [inline pkg/cmd/pipeline/start_test.go]: :+1:

File: pkg/cmd/pipeline/start.go (hunk 1 above)
Comment (vdemeester) [reply pkg/cmd/pipeline/start.go]: yeah `-r` doesn't exists

File: pkg/cmd/pipeline/start.go (hunk 2 above)
Comment (khrm) [reply pkg/cmd/pipeline/start.go]: Yes, because I use the remaining args to append if a user has given some passed some new args.

**Comment by hrishin** [comment]
@vdemeester final review? :)

File: pkg/cmd/pipeline/start.go [hunk 4]
```diff
@@ -200,6 +210,31 @@ func mergeParam(pr *v1alpha1.PipelineRun, optPar []string) error {
...
+		}
+	}
+
+	for _, v := range svcs {
+		pr.Spec.ServiceAccounts = append(pr.Spec.ServiceAccounts,
+			v)
```
Comment (vdemeester) [inline pkg/cmd/pipeline/start.go]: why the carriage return here ?

File: docs/cmd/tkn_pipeline_start.md [hunk 5]
```diff
@@ -36,11 +36,12 @@ two parameters (foo and bar)
...
-  -l, --last                    re-run the pipeline using last pipelinerun values
-  -p, --param strings           pass the param
-  -r, --resource strings        pass the resource name and ref
-  -s, --serviceaccount string   pass the serviceaccount name
+  -h, --help                      help for start
+  -l, --last                      re-run the pipeline using last pipelinerun values
+  -p, --param strings             pass the param
+  -r, --resource strings          pass the resource name and ref
+  -s, --serviceaccount string     pass the serviceaccount name
+      --serviceaccounts strings   pass the service account corresponding to the task
```
Comment (vdemeester) [inline docs/cmd/tkn_pipeline_start.md]: Should we use `task-serviceaccount` instead ? `--task-serviceaccount my-task=my-sa` is more user-friendly I think ? :thinking:

File: pkg/cmd/pipeline/start.go (hunk 4 above)
Comment (khrm) [reply pkg/cmd/pipeline/start.go]: By mistake. Resolved.

File: docs/cmd/tkn_pipeline_start.md (hunk 5 above)
Comment (khrm) [reply docs/cmd/tkn_pipeline_start.md]: Same here. I changed it.

**Comment by vdemeester** [comment]
/test all
Something is fishy with that prow update and already opened PR before it…

**Pull request merged**